app/models/file_search.rb

```diff
@@ -72,46 +94,59 @@ def year_range
     all_years
   end
 
-  def results
-    SearchResults.new(files)
-  end
-
   def files
-    params.empty? ? highlighted_files : filtered_files
+    query = {}
+    query[:q] = search_term if search_term
+    query[:f] = filters unless filters.empty?
+    query[:page] = page if page
+
+    (response, documents) = catalog_query.search_results(query, catalog_query.search_params_logic)
+    if @response = response
+      @total_items = @response.response["numFound"]
+      @total_pages = @response.total_pages
+      @current_page = @response.current_page
+      @next_page = @response.next_page
+    end
+    @files ||= file_query.find(documents.map(&:id))
   end
 
   PRIMARY_VENUES = ["Elizabethan", "Angus Bowmer", "Thomas", "The Green Show"]
   OTHER_VENUE = "Other"
-  RECORD_TYPES = %w[images videos audios articles]
+  RESOURCE_TYPES = %w[images videos audios articles]
 
   private
 
   attr_reader :params, :catalog_query, :file_query
 
+  def has_query_params?
+    (params.stringify_keys.keys & %w(q work venues years types)).any?
+  end
+
   def hardcoded_venues
     PRIMARY_VENUES.map { |name| ProductionCredits::Venue.find_by(name: name) }.compact.tap do |venues|
       venues << ProductionCredits::Venue.new(name: OTHER_VENUE)
     end
   end
 
-  def highlighted_files
-    file_query.where(highlighted: "1")
+  def filters
+    {}
+    .merge(work_filter)
+    .merge(venue_filter)
+    .merge(year_filter)
+    .merge(resource_type_filter)
+    .merge(highlight_filter)
```

Comment:
Formatting?  Atom messes this up, but I think the pipeline should be indented two spaces from the original collection